source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,134 @@ namespace aspect
 
     return (material_model->get_model_dependence().viscosity != MaterialModel::NonlinearDependence::none);
   }
-}
 
+
+
+  template <int dim>
+  void Simulator<dim>::apply_limiter_to_dg_solutions (LinearAlgebra::BlockVector &vector)
+  {
+    // create a quadrature formula based on the compositional element alone.
+    // be defensive about determining that a compositional field actually exists
+    if ( parameters.n_compositional_fields == 0)
+      AssertThrow (introspection.base_elements.compositional_fields
+                   != numbers::invalid_unsigned_int,
+                   ExcMessage("This function cannot be used without compositional fields."));
+
+    const QGauss<1> quadrature_formula_1(parameters.composition_degree+1);
+    const QGaussLobatto<1> quadrature_formula_2(parameters.composition_degree+1);
+    /**
+     * TODO: currently it only works for 2 dimentional problem. Will add the 3D special selected quadrature points and weight later.
```

Comment:
Please formalize this condition by adding an AssertThrow that checks that `(dim == 2)` and give it a useful error message. If this goes into the main ASPECT version, somebody **will** use your option in 3D without checking the code, and currently this might crash with a strange error message, or might even run and give wrong results.